pkg/kv/kvserver/replica_send.go

```diff
@@ -111,32 +113,19 @@ func (r *Replica) Send(
 func (r *Replica) SendWithWriteBytes(
 	ctx context.Context, ba roachpb.BatchRequest,
 ) (*roachpb.BatchResponse, *StoreWriteBytes, *roachpb.Error) {
-	return r.sendWithoutRangeID(ctx, &ba)
+	if r.store.cfg.Settings.CPUProfileType() == cluster.CPUProfileWithLabels {
+		prevCtx := ctx // Capture it for the defer below to pick the correct value.
```

Comment:
I think we don't even need prevCtx, see here:

https://play.golang.com/p/CHTIdiq7dNB

I still think it's worth keeping to avoid having future readers question the correctness of this change.

Reply:
Right, indeed: https://go.dev/ref/spec#Defer_statements.

I left a note.